ui/elements/Modal/Modal.tsx

```diff
@@ -120,81 +116,89 @@ const Modal = forwardRef<HTMLDivElement, ModalProps>((props, ref) => {
-                                        size="small"
-                                        variant="tertiary"
-                                        color="neutral"
-                                        icon="close"
-                                        ariaLabel="Close"
-                                        buttonProps={{ autoFocus: true }}
-                                    />
-                                </StyledModalAction>
-                            )}
-
-                            {banner && (
-                                <StyledModalBanner>{banner}</StyledModalBanner>
-                            )}
-
-                            {title && (
-                                <StyledModalTitle id={`modal-title-${id}`}>
-                                    {title}
-                                </StyledModalTitle>
-                            )}
-
-                            {description && (
-                                <StyledModalDescription
-                                    id={`modal-description-${id}`}
-                                >
-                                    {description}
-                                </StyledModalDescription>
+                        <StyledModal
+                            onClick={(e: React.MouseEvent<HTMLDivElement>) =>
+                                e.stopPropagation()
+                            }
+                            ref={modalRef}
+                            width={width}
+                            height={height}
+                            minHeight={minHeight}
+                            minWidth={minWidth}
+                            maxHeight={maxHeight}
+                            maxWidth={maxWidth}
+                            mobileBottomFullWidth={mobileBottomFullWidth}
+                            modalPadding={modalPadding}
+                            $border={border}
+                            aria-labelledby={`modal-title-${id}`}
+                            aria-describedby={`modal-description-${id}`}
+                            aria-modal="true"
+                            role="dialog"
+                            data-testid="testid-modal"
+                            {...motionProps}
+                        >
+                            <StyledModalMain $overflow={overflow}>
+                                {showCloseButton && (
+                                    <StyledModalAction justifyContent="end">
+                                        <IconButton
+                                            onClick={handleClose}
+                                            size="small"
+                                            variant="tertiary"
+                                            color="neutral"
+                                            icon="close"
+                                            ariaLabel="Close"
+                                            buttonProps={{ autoFocus: true }}
```

Comment:
would having autoFocus as true not conflict with the initialFocus property set above?